From 1798ac8390d35824d98f716736647ff03ea151b3 Mon Sep 17 00:00:00 2001
From: Jasper St. Pierre <jstpierre@mecheye.net>
Date: Fri, 04 Jan 2013 02:58:47 +0000
Subject: notification_area: Use the generic marshaller

The copy/paste code in gnome-shell has this change, so
make it here too as well.
---
diff --git a/applets/notification_area/Makefile.am b/applets/notification_area/Makefile.am
index 9734536..355737e 100644
--- a/applets/notification_area/Makefile.am
+++ b/applets/notification_area/Makefile.am
@@ -17,8 +17,6 @@ AM_CFLAGS = $(WARN_CFLAGS)
 libtray_la_SOURCES =		\
 	fixedtip.h		\
 	fixedtip.c		\
-	na-marshal.c		\
-	na-marshal.h		\
 	na-tray.c		\
 	na-tray.h		\
 	na-tray-child.c		\
@@ -61,15 +59,6 @@ notification_area_applet_LDADD = $(NOTIFICATION_AREA_LDADD)
 notification_area_applet_CFLAGS = $(AM_CFLAGS)
 endif
 
-na-marshal.h: na-marshal.list $(GLIB_GENMARSHAL)
-	$(AM_V_GEN)$(GLIB_GENMARSHAL) $< --header --prefix=_na_marshal > $@
-
-na-marshal.c: na-marshal.list $(GLIB_GENMARSHAL)
-	$(AM_V_GEN)echo "#include \"na-marshal.h\"" > $@ && \
-	$(GLIB_GENMARSHAL) $< --body --prefix=_na_marshal >> $@
-
-BUILT_SOURCES = na-marshal.c na-marshal.h
-
 appletdir       = $(appletsdir)
 applet_in_files = org.gnome.panel.NotificationAreaApplet.panel-applet.in
 applet_DATA     = $(applet_in_files:.panel-applet.in=.panel-applet)
@@ -97,8 +86,7 @@ endif
 EXTRA_DIST =								\
 	org.gnome.panel.NotificationAreaApplet.panel-applet.in.in	\
 	$(ui_DATA)							\
-	$(service_in_files)						\
-	na-marshal.list
+	$(service_in_files)
 
 CLEANFILES =			\
 	$(applet_DATA) 		\
diff --git a/applets/notification_area/fixedtip.c b/applets/notification_area/fixedtip.c
index 0e3c217..861e4ab 100644
--- a/applets/notification_area/fixedtip.c
+++ b/applets/notification_area/fixedtip.c
@@ -91,8 +91,7 @@ na_fixed_tip_class_init (NaFixedTipClass *class)
 		  G_OBJECT_CLASS_TYPE (class),
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (NaFixedTipClass, clicked),
-		  NULL, NULL,
-		  g_cclosure_marshal_VOID__VOID,
+		  NULL, NULL, NULL,
 		  G_TYPE_NONE, 0);
 
   g_type_class_add_private (class, sizeof (NaFixedTipPrivate));
diff --git a/applets/notification_area/na-marshal.list b/applets/notification_area/na-marshal.list
deleted file mode 100644
index e3fc399..0000000
--- a/applets/notification_area/na-marshal.list
+++ /dev/null
@@ -1,3 +0,0 @@
-VOID:OBJECT,OBJECT
-VOID:OBJECT,STRING,LONG,LONG
-VOID:OBJECT,LONG
diff --git a/applets/notification_area/na-tray-manager.c b/applets/notification_area/na-tray-manager.c
index da98f00..8999350 100644
--- a/applets/notification_area/na-tray-manager.c
+++ b/applets/notification_area/na-tray-manager.c
@@ -34,8 +34,6 @@
 #endif
 #include <gtk/gtk.h>
 
-#include "na-marshal.h"
-
 /* Signals */
 enum
 {
@@ -146,8 +144,7 @@ na_tray_manager_class_init (NaTrayManagerClass *klass)
 		  G_OBJECT_CLASS_TYPE (klass),
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (NaTrayManagerClass, tray_icon_added),
-		  NULL, NULL,
-		  g_cclosure_marshal_VOID__OBJECT,
+		  NULL, NULL, NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_SOCKET);
 
@@ -156,8 +153,7 @@ na_tray_manager_class_init (NaTrayManagerClass *klass)
 		  G_OBJECT_CLASS_TYPE (klass),
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (NaTrayManagerClass, tray_icon_removed),
-		  NULL, NULL,
-		  g_cclosure_marshal_VOID__OBJECT,
+		  NULL, NULL, NULL,
 		  G_TYPE_NONE, 1,
 		  GTK_TYPE_SOCKET);
   manager_signals[MESSAGE_SENT] =
@@ -165,8 +161,7 @@ na_tray_manager_class_init (NaTrayManagerClass *klass)
 		  G_OBJECT_CLASS_TYPE (klass),
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (NaTrayManagerClass, message_sent),
-		  NULL, NULL,
-		  _na_marshal_VOID__OBJECT_STRING_LONG_LONG,
+		  NULL, NULL, NULL,
 		  G_TYPE_NONE, 4,
 		  GTK_TYPE_SOCKET,
 		  G_TYPE_STRING,
@@ -177,8 +172,7 @@ na_tray_manager_class_init (NaTrayManagerClass *klass)
 		  G_OBJECT_CLASS_TYPE (klass),
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (NaTrayManagerClass, message_cancelled),
-		  NULL, NULL,
-		  _na_marshal_VOID__OBJECT_LONG,
+		  NULL, NULL, NULL,
 		  G_TYPE_NONE, 2,
 		  GTK_TYPE_SOCKET,
 		  G_TYPE_LONG);
@@ -187,8 +181,7 @@ na_tray_manager_class_init (NaTrayManagerClass *klass)
 		  G_OBJECT_CLASS_TYPE (klass),
 		  G_SIGNAL_RUN_LAST,
 		  G_STRUCT_OFFSET (NaTrayManagerClass, lost_selection),
-		  NULL, NULL,
-		  g_cclosure_marshal_VOID__VOID,
+		  NULL, NULL, NULL,
 		  G_TYPE_NONE, 0);
 
 #if defined (GDK_WINDOWING_X11)
--
cgit v0.9.1
